cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -126,6 +126,33 @@ public void testStopBeforeStart() throws RuntimeException {
   }
 
   @Test
+  public void testDeleteSuspendedWorkflow() {
+    Id.Namespace namespaceId = new Id.Namespace("namespace1");
+
+    // Test delete application
+    Id.Application appId1 = new Id.Application(namespaceId, "app1");
+    Id.Program programId1 = Id.Program.from(appId1, ProgramType.WORKFLOW, "pgm1");
+    RunId run1 = RunIds.generate();
+    store.setStart(programId1, run1.getId(), runIdToSecs(run1));
+    store.setSuspend(programId1, run1.getId());
+    store.removeApplication(appId1);
+    Assert.assertEquals(0, store.getRuns(programId1, ProgramRunStatus.ALL, 0,
```

Comment:
@chtyim One thing to note then, is that if it fails, you then have to debug further to see what the actual size was, if using `isEmpty()`.